### PR TITLE
Fix Normalize ignoring a single explicit in_min/in_max bound

### LINKED ISSUE #1509: Normalize silently ignores in_min or in_max when only one is provided
### Description

`Normalize` ignores an explicitly supplied `in_min` when `in_max` is `None`, and vice versa.
Both bounds are calculated from the input data instead.

The parameter documentation describes automatic calculation when the corresponding bound is `None`, so I would expect an explicitly supplied bound to be preserved.

### Reproduction

```python
import torch
import torchio as tio

image = tio.ScalarImage(
    torch.tensor([0., 50., 100.]).reshape(1, 1, 1, 3)
)

for bounds in ({"in_min": 50.}, {"in_max": 50.}):
    result = tio.Normalize(
        out_min=0.,
        out_max=1.,
        **bounds,
    )(image)
    print(bounds, result.data.flatten().tolist())
```

### Actual behavior

```text
{'in_min': 50.0} [0.0, 0.5, 1.0]
{'in_max': 50.0} [0.0, 0.5, 1.0]
```

### Expected behavior

```text
{'in_min': 50.0} [0.0, 0.0, 1.0]
{'in_max': 50.0} [0.0, 1.0, 1.0]
```

### Possible cause

`Normalize.make_params()` uses the explicit bounds only when both are non-`None`:

```python
if self.in_min is not None and self.in_max is not None:
```

Otherwise, it computes both bounds from percentiles.
Each bound could be resolved independently, calculating only the missing one.

### Version

Reproduced on `main`, commit `2b019d2a9659c838408c73ae77327ac5eb87b190`, TorchIO `2.0.0a2`, using synthetic CPU tensors.

### PR DESCRIPTION
**Pull request by eminesu**
## Summary

`Normalize` ignores an explicitly supplied `in_min` when `in_max` is `None`, and vice versa — the explicit value is silently discarded and **both** bounds are computed from the data percentiles instead. This happens because `make_params` only uses the explicit bounds when *both* are provided.

This PR resolves each input bound independently: an explicitly supplied bound is always honored, and only the missing one is computed from the percentiles. Behaviour when both bounds are given, or when both are `None`, is unchanged.

## Reproduction (before this PR)

```python
import torch
import torchio as tio

image = tio.ScalarImage(torch.tensor([0., 50., 100.]).reshape(1, 1, 1, 3))
print(tio.Normalize(out_min=0., out_max=1., in_min=50.)(image).data.flatten().tolist())
# before: [0.0, 0.5, 1.0]   (in_min=50 ignored)
# after:  [0.0, 0.0, 1.0]
```

## Changes

- `Normalize.make_params`: resolve `in_min` / `in_max` independently and override the auto-computed bound(s) with any explicit value.
- Add regression tests for the `in_min`-only and `in_max`-only cases.

Fixes #1509

### CODE REVIEW

File: src/torchio/transforms/intensity/normalize.py [hunk 1]
```diff
@@ -122,22 +122,31 @@ def make_params(self, batch: SubjectsBatch) -> dict[str, Any]:
...
             for name, img_batch in self._get_images(batch).items():
                 mask = self._get_mask(img_batch, batch)
-                in_ranges[name] = _percentile_range(
+                in_min, in_max = _percentile_range(
                     img_batch.data[0],
                     mask,
                     pct_low,
                     pct_high,
                     name,
                 )
```
[the remark this answers is not in the document]
Comment (fepegar) [reply src/torchio/transforms/intensity/normalize.py:143]: @eminesu would you like to address this?
